From 26243c837cafd36484c1bed5978afff631650eaa Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Thu, 21 Oct 2021 10:59:18 +0000
Subject: IWYU: missing include for numeric limits used in partition alloc
 constants.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix build:
    ../../base/allocator/partition_allocator/partition_alloc_constants.h:355:43: error: ‘numeric_limits’ is not a member of ‘std’
      355 | constexpr size_t kMaxFreeableSpans = std::numeric_limits<int8_t>::max();
          |                                           ^~~~~~~~~~~~~~
    ../../base/allocator/partition_allocator/partition_alloc_constants.h:355:64: error: expected primary-expression before ‘>’ token
      355 | constexpr size_t kMaxFreeableSpans = std::numeric_limits<int8_t>::max();
          |                                                                ^
    ../../base/allocator/partition_allocator/partition_alloc_constants.h:355:67: error: ‘::max’ has not been declared; did you mean ‘std::max’?
      355 | constexpr size_t kMaxFreeableSpans = std::numeric_limits<int8_t>::max();
          |                                                                   ^~~
          |                                                                   std::max

Bug: 957519
Change-Id: I0a2f839589650901c5abf1feae5f3a3ef18fefc6
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3231561
Reviewed-by: Benoit L <lizeb@chromium.org>
Commit-Queue: José Dapena Paz <jdapena@igalia.com>
Cr-Commit-Position: refs/heads/main@{#933811}
---
 base/allocator/partition_allocator/partition_alloc_constants.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/base/allocator/partition_allocator/partition_alloc_constants.h b/base/allocator/partition_allocator/partition_alloc_constants.h
index b1309004b79f5..cdcf87b3bbd4d 100644
--- a/base/allocator/partition_allocator/partition_alloc_constants.h
+++ b/base/allocator/partition_allocator/partition_alloc_constants.h
@@ -9,6 +9,7 @@
 #include <cstddef>
 
 #include <algorithm>
+#include <limits>
 
 #include "base/allocator/partition_allocator/address_pool_manager_types.h"
 #include "base/allocator/partition_allocator/page_allocator_constants.h"
-- 
2.32.0

